lib/puppet/face/module/generate.rb

```diff
@@ -1,256 +1,14 @@
-    description <<-EOT
-      Generates boilerplate for a new module by creating the directory
-      structure and files recommended for the Puppet community's best practices.
-
-      A module may need additional directories beyond this boilerplate
-      if it provides plugins, files, or templates.
-    EOT
-
-    returns _("Array of Pathname objects representing paths of generated files.")
-
-    examples <<-EOT
-      Generate a new module in the current directory:
-
-      $ puppet module generate puppetlabs-ssh
-      We need to create a metadata.json file for this module.  Please answer the
-      following questions; if the question is not applicable to this module, feel free
-      to leave it blank.
-
-      Puppet uses Semantic Versioning (semver.org) to version modules.
-      What version is this module?  [0.1.0]
-      -->
-
-      Who wrote this module?  [puppetlabs]
-      -->
-
-      What license does this module code fall under?  [Apache-2.0]
-      -->
-
-      How would you describe this module in a single sentence?
-      -->
-
-      Where is this module's source code repository?
-      -->
-
-      Where can others go to learn more about this module?
-      -->
-
-      Where can others go to file issues about this module?
-      -->
-
-      ----------------------------------------
-      {
-        "name": "puppetlabs-ssh",
-        "version": "0.1.0",
-        "author": "puppetlabs",
-        "summary": null,
-        "license": "Apache-2.0",
-        "source": "",
-        "project_page": null,
-        "issues_url": null,
-        "dependencies": [
-          {
-            "name": "puppetlabs-stdlib",
-            "version_requirement": ">= 1.0.0"
-          }
-        ]
-      }
-      ----------------------------------------
-
-      About to generate this metadata; continue? [n/Y]
-      -->
-
-      Notice: Generating module at /Users/username/Projects/puppet/puppetlabs-ssh...
-      Notice: Populating ERB templates...
-      Finished; module generated in puppetlabs-ssh.
-      puppetlabs-ssh/manifests
-      puppetlabs-ssh/manifests/init.pp
-      puppetlabs-ssh/metadata.json
-      puppetlabs-ssh/README.md
-      puppetlabs-ssh/spec
-      puppetlabs-ssh/spec/spec_helper.rb
-      puppetlabs-ssh/tests
-      puppetlabs-ssh/tests/init.pp
-    EOT
-
-    option "--skip-interview" do
-      summary _("Bypass the interactive metadata interview")
-      description <<-EOT
-        Do not attempt to perform a metadata interview.  Primarily useful for automatic
-        execution of `puppet module generate`.
-      EOT
-    end
-
-    arguments _("<name>")
+    #TRANSLATORS 'Puppet Development Kit' is the name of the software package replacing this action and should not be translated.
+    description _("This action has been replaced by Puppet Development Kit. For more information visit https://puppet.com/docs/pdk/latest/pdk.html.")
 
     when_invoked do |name, options|
```

Comment:
Since this is going to cause an error message no matter what is provided to it, it might be worth removing the restriction that it's given a single argument?

```
% be ./bin/puppet module generate
Error: puppet module generate takes 1 argument, but you gave 0
Error: Try 'puppet help module generate' for usage
```

Maybe changing the `when_invoked` line to `when_invoked do |*args|`, so it'll give the deprecation message no matter how many arguments it's given (`puppet module build` is already like this):

```
% be ./bin/puppet module generate
Error: This action has been replaced by Puppet Development Kit. For more information visit https://puppet.com/docs/pdk/latest/pdk.html.
Error: Try 'puppet help module generate' for usage
```

Reply:
good idea. fixed.